### DISCUSSION TITLE
Application does not work in mobile devices (iPhone, iPad)

### DISCUSSION THREAD

**Discussion by exanubes** [vercel/next.js#48292] [Help] [answered]
### Summary

I've built a small site with form submissions. It works fine on desktop but mobile - iphone & ipad - does not work properly.

The site renders just fine, however, after filling in any field, it shows "required" error as if the field was still empty.

Upon removing the validation from the fields. The form never submits, it just hangs indefinitely.

None of these issues occur in desktop and I haven't tested any android devices.

Has anyone encountered a problem like this before?

### Mobile - with validation schema

https://user-images.githubusercontent.com/83121511/231481532-012fe44b-6d6e-416c-ae92-a5b5c290e426.mov

### Mobile - submitting form without schema

https://user-images.githubusercontent.com/83121511/231481570-467cfbc4-c057-45f5-8916-5379c88af03c.mov

### Desktop - everything works fine

https://user-images.githubusercontent.com/83121511/231482720-c6323e46-90f8-4549-ab15-07a3447f597a.mov

### Additional information

```js
### Next info


    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.14.0
      npm: 9.3.1
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.1.1
      eslint-config-next: 13.1.1
      react: 18.2.0
      react-dom: 18.2.0
```
```


### Example

_No response_

**Comment by ciruz** [comment]
Do you have live url to try it myself and maybe some code to look what's going on?

**Comment by exanubes** [comment, marked as answer]
Hi @ciruz, thanks for replying. I've managed to narrow it down to certain regular expressions and it seems as though it is quite common for apple os's to break because of them, in case anyone else has this issue at some point :).

This is the troublesome code:

```typescript
          const emailRegex = /^([a-zA-Z0-9_.-])+@(([a-zA-Z0-9-])+\.)+([a-zA-Z0-9]{2,4})+$/;
          const phoneRegex = /(?<!\w)(\(?(\+|00)?48\)?)?[ -]?\d{3}[ -]?\d{3}[ -]?\d{3}(?!\w)/;
```
